### PR TITLE
NATS Streaming output plugin

### PR DESCRIPTION
**Pull request by RashadAnsari**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

File: plugins/outputs/stan/stan.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	}
+
+	if s.Secure {
+		tlsConfig, err := s.ClientConfig.TLSConfig()
+		if err != nil {
+			return err
+		}
+
+		natsOpts = append(natsOpts, nats.Secure(tlsConfig))
+	}
```
Comment (ssoroka) [inline plugins/outputs/stan/stan.go:92]: All the setup stuff belongs in an `Init() error` function, you don't want it repeated each connect.

File: plugins/outputs/stan/stan_test.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	}
+
+	// Verify that we can connect to the NATS Streaming daemon
+	err := n.Connect()
+	require.NoError(t, err)
+
+	// Verify that we can successfully write data to the NATS Streaming daemon
+	err = n.Write(testutil.MockMetrics())
+	require.NoError(t, err)
+}
```
Comment (ssoroka) [inline plugins/outputs/stan/stan_test.go:33]: I'd like to see some kind of -short test, otherwise we'll likely not catch it if this breaks.

**Review by ssoroka — CHANGES_REQUESTED** [review]
looks pretty good. Would like to see more tests for it.

**Comment by ssoroka** [comment]
Looks like there's some package versioning issues.

**Comment by Kuzyashin** [comment]
Any updates here? How can I help speed up approving this PR ?

File: plugins/outputs/stan/stan_test.go (hunk 2 above)
Comment (Kuzyashin) [reply plugins/outputs/stan/stan_test.go:33]: This tests looks like tests for other outputs.

**Comment by sspaink** [comment]
It looks like NATS Streaming service is deprecated (https://github.com/nats-io/nats-streaming-server) is there still a reason for this plugin? It seems it was replaced by a service called Jetstream.

**Comment by 1995parham** [comment]
@sspaink Yes, I think it would be great if we can replace this plugin by jetstream which is based on nats.

**Comment by sspaink** [comment]
Ok great I will go ahead and close this PR for now then, and if anyone would like to submit a Jetstream plugin pull request we can review it.

**Pull request closed without merging**